frontend/src/app/_services/route.service.ts

```diff
@@ -0,0 +1,285 @@
+import {Injectable} from '@angular/core';
+import {BackendPath, FrontendPath, NavigationParams, NavigationRoute} from '../_models/route';
+import {MenuHierarchy} from '../_models/menu';
+
+const EMPTY_CHAR = '_';
+const EMPTY_CHAR_REGEX = /_/g;
+export const NAVIGATE_PATH = 'app/documentation/navigate/';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class RouteService {
+
+    constructor() {
+    }
+
+    static legacyFullFrontEndUrlToFullFrontEndUrl(input: string): string {
+        if (input == undefined || input.length == 0) {
+            return input;
+        }
+        const commaSplit = input.split(';');
+        if (commaSplit.length != 2) {
+            return input;
+        }
+        const fullFrontEndUrlWithNodes = commaSplit[0];
+        const pathPart = commaSplit[1];
+        const equalSplit = pathPart.split('=');
+        if (equalSplit.length != 2) {
+            return input;
+        }
+        const legacyUrl = pathPart.split('=')[1];
+        let doubleChevronSplit = legacyUrl.split('>>');
+        if (doubleChevronSplit.length == 1 ) {
+           doubleChevronSplit = legacyUrl.split('%3E%3E');
+        }
+
+        let project = '';
+        let branch = '';
+        let directoriesAndPage = '';
+        if (doubleChevronSplit.length == 2) {
+            project = doubleChevronSplit[0];
+            branch = EMPTY_CHAR;
+            directoriesAndPage = doubleChevronSplit[1];
+        } else {
+            let chevronSplit = legacyUrl.split('>');
+            if (chevronSplit.length == 1 ) {
+                chevronSplit = legacyUrl.split('%3E');
+            }
+            if (chevronSplit.length == 3) {
+                project = chevronSplit[0];
+                branch = chevronSplit[1];
+                directoriesAndPage = chevronSplit[2];
+            } else {
+                return input;
+            }
+        }
+        const slashSplit = directoriesAndPage.split('/');
+        if (slashSplit.length != 2) {
+            return input;
+        }
+        let directories = slashSplit[0];
+        if (directories.length > 1) {
+            directories = directories.substr(0, directories.length - 1);
+        }
+        const page = slashSplit[1];
+
+        return `${fullFrontEndUrlWithNodes}/${project}/${branch}/${directories}/${page}`;
```

Comment:
This method is really long. Could it be splitted in several sub methods, so that we better understand what it does ?

Reply:
It's well tested and it will be removed once the legacy path won't be supported... so I won't bother ;)